src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -39,6 +39,34 @@ using namespace Antares;
 using namespace Antares::Data;
 using namespace Yuni;
 
+static void importShortTermStorages(
```

Comment:
We could take advantage of this function to compute the total number of ST storage clusters and store it into `problem.NumberOfShortTermStorages`
We would have : 
```C++
static void importShortTermStorages(const AreaList& areas, PROBLEME_HEBDO& problem)
{
    ...
}
```
Doing this would allow to remove the computation of this number from **runtime.cpp**, and even remove all references to ST storage from class **StudyRuntimeInfos**, which I don't like because it stores all kinds of things, don't store proper runtime data and is probably useless eventually.

If we decide to make this move, we'll have to call function **importShortTermStorages** before function **SIM_AllocationProblemeHebdo** : 
```C++
void SIM_InitialisationProblemeHebdo(Data::Study& study,
                                     PROBLEME_HEBDO& problem,
                                     int NombreDePasDeTemps,
                                     uint numSpace)
{
    ...
    importShortTermStorages(study.areas, problem);
    SIM_AllocationProblemeHebdo(problem, NombreDePasDeTemps);
    ...
}
```

Reply:
I agree, but having `shortTermStorageCount` in runtime.h allows to display this log (runtime.cpp)
```cpp
    logs.info() << "     short-term storages: " << shortTermStorageCount;
```

I'm not sure we could display it so easily if it was only in `PROBLEME_HEBDO`.